### PR TITLE
Add tests for InfoboxNode.

### PR DESCRIPTION
**Pull request by cassierecher**

### CODE REVIEW

File: claat/nodes/infobox_test.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+	"github.com/google/go-cmp/cmp"
+)
+
+// AllowUnexported option for cmp to make sure we can diff properly.
+var cmpOpt = cmp.AllowUnexported(InfoboxNode{}, node{}, ListNode{}, TextNode{})
+
+func TestNewInfoboxNode(t *testing.T) {
+	tests := []struct {
+		name      string
+		inKind    InfoboxKind
```
Comment (DragonRider0o0) [inline claat/nodes/infobox_test.go:15]: Consider dropping the `in` prefix; `kind` and `content` are sufficient.

File: claat/nodes/infobox_test.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+			// TODO: Should we set a default value?
+			name:      "NoKind",
+			inContent: []Node{NewTextNode("orange")},
+			out: &InfoboxNode{
+				node:    node{typ: NodeInfobox},
+				Content: NewListNode(NewTextNode("orange")),
+			},
+		},
+		{
+			// TODO: Should a list of exactly one list be flattened?
```
Comment (DragonRider0o0) [inline claat/nodes/infobox_test.go:89]: I don't think we should flatten single item lists, I think that's best left as an author decision rather than a platform decision.

File: claat/nodes/infobox_test.go [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+			inKind:    InfoboxNegative,
+			inContent: []Node{NewTextNode("orange"), NewTextNode("strawberry"), NewTextNode("pineapple")},
+			out: &InfoboxNode{
+				node:    node{typ: NodeInfobox},
+				Kind:    InfoboxNegative,
+				Content: NewListNode(NewTextNode("orange"), NewTextNode("strawberry"), NewTextNode("pineapple")),
+			},
+		},
+		{
+			// TODO: Should we set a default value?
```
Comment (DragonRider0o0) [inline claat/nodes/infobox_test.go:80]: We should probably set the default value to positive.

File: claat/nodes/infobox_test.go [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+		inContent []Node
+		out       *InfoboxNode
+	}{
+		{
+			name:   "PositiveEmpty",
+			inKind: InfoboxPositive,
+			out: &InfoboxNode{
+				node: node{typ: NodeInfobox},
+				Kind: InfoboxPositive,
+				// TODO: Do we really want this to not be nil?
```
Comment (DragonRider0o0) [inline claat/nodes/infobox_test.go:25]: I think it's acceptable as is because we're calling the NewListNode's empty method when we evaluate empty.

File: claat/nodes/infobox_test.go [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+				Content: NewListNode(NewTextNode("orange"), NewTextNode("strawberry"), NewTextNode("pineapple")),
+			},
+		},
+		{
+			name:   "NegativeEmpty",
+			inKind: InfoboxNegative,
+			out: &InfoboxNode{
+				node: node{typ: NodeInfobox},
+				Kind: InfoboxNegative,
+				// TODO: Do we really want this to not be nil?
```
Comment (DragonRider0o0) [inline claat/nodes/infobox_test.go:55]: I think it's acceptable as is because we're calling the NewListNode's empty method when we evaluate empty.

File: claat/nodes/infobox_test.go [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+			inContent: []Node{NewTextNode("orange"), NewTextNode("strawberry"), NewTextNode("pineapple")},
+			out: &InfoboxNode{
+				node:    node{typ: NodeInfobox},
+				Kind:    InfoboxNegative,
+				Content: NewListNode(NewTextNode("orange"), NewTextNode("strawberry"), NewTextNode("pineapple")),
+			},
+		},
+		{
+			// TODO: Should we set a default value?
+			name:      "NoKind",
```
Comment (DragonRider0o0) [inline claat/nodes/infobox_test.go:81]: We should probably add a test for an unsupported kind as well.

File: claat/nodes/infobox_test.go [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			out := NewInfoboxNode(tc.inKind, tc.inContent...)
+			if diff := cmp.Diff(tc.out, out, cmpOpt); diff != "" {
+				t.Errorf("NewInfoboxNode(%q, %v) got diff (-want +got): %s", tc.inKind, tc.inContent, diff)
+				return
+			}
+		})
+	}
+}
```
Comment (DragonRider0o0) [inline claat/nodes/infobox_test.go:109]: Add a `TestInfoboxNodeEmpty` test to validate `Empty`.

File: claat/nodes/infobox_test.go (hunk 2 above)
Comment (cassierecher) [reply claat/nodes/infobox_test.go:89]: It would just be flattening in the case of a list containing exactly one list. I'm not sure it matters in terms of output; this may be an optimization we can make before rendering.

File: claat/nodes/infobox_test.go (hunk 3 above)
Comment (cassierecher) [reply claat/nodes/infobox_test.go:80]: Probably, but want to leave this for a future PR and have this one just contain tests for the code as it is today.

File: claat/nodes/infobox_test.go (hunk 5 above)
Comment (cassierecher) [reply claat/nodes/infobox_test.go:55]: I think this eventually ties in to a look at our node system. I'm not sure we have a universal standard over whether an empty node is semantically synonymous to the lack of a node or not.

File: claat/nodes/infobox_test.go (hunk 4 above)
Comment (cassierecher) [reply claat/nodes/infobox_test.go:25]: We could just as easily check if Content is nil, or perhaps in addition to. Would like to defer this to a separate PR

File: claat/nodes/infobox_test.go (hunk 1 above)
Comment (cassierecher) [reply claat/nodes/infobox_test.go:15]: I think this makes it more readable, it provides more cues to what the test's inputs and outputs are.

File: claat/nodes/infobox_test.go (hunk 4 above)
Comment (DragonRider0o0) [reply claat/nodes/infobox_test.go:25]: SGTM.

File: claat/nodes/infobox_test.go (hunk 5 above)
Comment (DragonRider0o0) [reply claat/nodes/infobox_test.go:55]: Fair point.

File: claat/nodes/infobox_test.go (hunk 3 above)
Comment (DragonRider0o0) [reply claat/nodes/infobox_test.go:80]: SGTM

File: claat/nodes/infobox_test.go (hunk 2 above)
Comment (DragonRider0o0) [reply claat/nodes/infobox_test.go:89]: Right, but the first list represents the content of the infobox while the second list represents a list within the content of the infobox.

Comment (cassierecher) [reply claat/nodes/infobox_test.go:89]: Ah, I understand now. Will remove this

File: claat/nodes/infobox_test.go (hunk 6 above)
Comment (cassierecher) [reply claat/nodes/infobox_test.go:81]: Added

File: claat/nodes/infobox_test.go (hunk 7 above)
Comment (cassierecher) [reply claat/nodes/infobox_test.go:126]: Added

File: claat/nodes/infobox_test.go (hunk 1 above)
Comment (cassierecher) [reply claat/nodes/infobox_test.go:15]: Parallel discussion on https://github.com/googlecodelabs/tools/pull/596#discussion_r666499324

**Review by shawnbuso — APPROVED** [review]
